### DISCUSSION TITLE
一个bug和一些功能请求

### DISCUSSION THREAD

**Discussion by chenxingmoonset** [xiaohucode/yidaRule#172] [Q&A]
bug：安卓端v1.1.26  开启文本朗读后熄屏/进入后台运行，读完当前段落会自动停止，需要点亮屏幕/打开app后继续阅读。但是再次熄屏/后台运行仍会停止朗读。系统：HarmonyOS 2.0.0

功能请求：
1、startBrowser可否添加response body的返回值
2、可不可以添加漫画的缓存和导出😍
3、js中可否添加jsonpath解析支持(不知道loadJS是否支持远程js链接引入，貌似可以直接添加到loadJS里？如果没必要就算了)

ps：话说现在益达支持外部tts吗

最后，炮佬牛逼，益达加油！！

**Comment by mabDc** [comment]
ps在原本的亦搜就有的。也支持http链接载入js。
使用方法是`require(链接)`，类似现在流行的`import`。
第3点应该不难，和xpath大同小异，我觉得实现是没差别，参考源码：

https://github.com/mabDc/eso/blob/xiaohu/lib/api/api_js_engine.dart#L113

https://github.com/mabDc/eso/blob/xiaohu/lib/api/api_js_engine.dart#L133

```js

    await setToGlobalObject.invoke([
      "require",
      IsolateFunction((dynamic url) async {
        if (url == null) return null;
        final module = url.toString();
        if (module.startsWith("http")) {
          final res = await http.get(Uri.parse(module));
          return await _engine.evaluate(res.body + ";0;");
        } else {
          try {
            final js = await rootBundle.loadString("lib/assets/" +
                module.replaceFirst(new RegExp(r".js$"), "") +
                ".js");
            return await _engine.evaluate(js + ";0;");
          } catch (e) {
            return null;
          }
        }
      }),
    ]);
    await setToGlobalObject.invoke([
      "xpath",
      IsolateFunction((String html, String xpath) async {
        return AnalyzerXPath().parse(html).getString(xpath);
      }),
    ]);
```

**Reply by chenxingmoonset** [reply]
👌
话说亦搜有startBrowser和getVerificationCode的支持计划嘛哈哈🤪

**Reply by mabDc** [reply]
居然用过哦，挺意外。有计划。但是我好久好久不写了。我现在感觉东西是二三年前的，很遥远。我看到阅读3也很陌生，不过阅读3确实功能多。

（）信我更新。母猪会上树。

**Reply by chenxingmoonset** [reply]
> 居然用过哦，挺意外。有计划。但是我好久好久不写了。我现在感觉东西是二三年前的，很遥远。我看到阅读3也很陌生，不过阅读3确实功能多。

哈哈，我算是个在益达，亦搜和阅读引领下初窥门径的源作者。接触源码之后感受更深刻了。支持大佬的进一步开发😍

**Reply by mabDc** [reply]
冷知识：亦搜与阅读3大约同时开工，甚至还早一点时候出成品。益达毕竟新品，继续努力吧。测试下，bug有部分。功能也简洁。
